index.js

```diff
@@ -2,6 +2,7 @@
 const mimicFn = require('mimic-fn');
 const isPromise = require('p-is-promise');
 
+const fnStore = new WeakMap();
```

Comment:
I think this could use a succinct comment about its purpose, so it's clearer when you read the code that it's for caching calls to mem itself.